test/e2e/handler/broken_configuration_test.go

```diff
@@ -0,0 +1,171 @@
+package handler
+
+import (
+	nmstate "github.com/nmstate/kubernetes-nmstate/api/shared"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+func createInterfaceWithInvalidName() nmstate.State {
+	return nmstate.NewState(`interfaces:
+  - name: eth666
+    type: ethernet
+    state: up`)
+}
+
+func createInterfaceWithInvalidField() nmstate.State {
+	return nmstate.NewState(`interfaces:
+  - name: eth1
+    type: ethernet
+    stasdte: up`)
+}
+
+func createInterfaceWithPingFail() nmstate.State {
+	return nmstate.NewState(`interfaces:
+  - name: eth0
+    type: ethernet
+    state: up
+    ipv4:
+      address:
+      - ip: "192.168.45.33"
+        prefix-length: 24
+      dhcp: false
+      enabled: true`)
+}
+
+var _ = FDescribe("NodeNetworkState", func() {
+	var (
+		DefaultNetworkBroken = "default-network-broken"
+
+		failMessages = [...]string{
+			"DEBUG    Async action: Create checkpoint started",
+			"DEBUG    Checkpoint None created for all devices",
+			"Traceback (most recent call last):",
+			"DEBUG    Nispor: current network state",
+			"WARNING  libnm version",
+			"rolling back desired state configuration: failed running probes after network changes: ",
+			"failed running probe 'ping' with after network reconfiguration -> currentState:",
+			"warnings.warn(",
+		}
+	)
+	Context("with invalid field", func() {
+		var (
+			messagesToKeep = [...]string{
+				"libnmstate.error.NmstateVerificationError:",
+				"desired",
+			}
+		)
+		BeforeEach(func() {
+			By("Creating the policy")
+			err := setDesiredStateWithPolicyAndNodeSelector(DefaultNetworkBroken, createInterfaceWithInvalidField(), map[string]string{"kubernetes.io/hostname": "node01"})
```

Comment:
You could define the nodeSelector variable within the Describe block, as it's the same for each test.
Also, please pick the node as `nodes[0]`, which is the first worker node in the cluster.
Running these policies on master may cause unintended trouble.

Reply:
Exactly this doesn't work, I'm not able to use `nodes[0]` in `Describe` block. 
However, I extracted this code into a helper function, so so the duplicity was removed anyway.